### PR TITLE
Make a new wayland backend using smithay-client-toolkit

### PR DESCRIPTION
**Pull request by DJMcNab**
Previous commit histories can be found in [waylander](https://github.com/djmcnab/glazier/tree/waylander) and [waylandest](https://github.com/djmcnab/glazier/tree/waylandest). Those histories were abandoned to rebase over #103, #104 and #105.

Works towards #28

Future work, which should probably be done before merge
- [ ] Input types (Mouse/Keyboard)

Other future work, which can be collaborated on once merged:
- [ ] Timers
- [ ] Proper invalidation
- [ ] Pop-up windows
- [ ] Solve client side decorations. Either API for it and require users to implement, or using a fallback, if can be made to work.
- [ ] Menus of all kinds - same as client side decorations
- [ ] Clipboard - our API is synchronous, but Wayland's is async
- [ ] Monitors - should access the `OutputState`, I think
       What do we actually use monitors for?
- [x] Text fields/compose
- [ ] Cursors
- [ ] XDG desktop portal (file open/save, etc.)
- [ ] Fractional scaling (needs changes in sctk)
- [ ] Work out how to customise the App ID

### CODE REVIEW

File: src/backend/wayland/application.rs
Comment (i509VCB) [inline src/backend/wayland/application.rs:64]: You could keep the `Connection` around for this or get the underlying `Backend` from wayland-backend and keep that.

File: src/backend/wayland/application.rs
Comment (i509VCB) [inline src/backend/wayland/application.rs:176]: I'd actually wake the event loop from the signal as well since you may need to wait for some more events to come in if this is done just before the event loop suspends the process.

File: src/backend/wayland/input/keyboard/mmap.rs
Comment (i509VCB) [inline src/backend/wayland/input/keyboard/mmap.rs:10]: Could we not use the memmap2 crate?

Comment (DJMcNab) [reply src/backend/wayland/input/keyboard/mmap.rs:10]: That is the plan, via the rust-x-bindings crate

However, that may require changes in the X11 backend as well, so I'm intending to include that (+ composing) in a follow up PR.

Comment (runiq) [reply src/backend/wayland/input/keyboard/mmap.rs:10]: I though the X11 backend was going away as well, or is this a kind of 'don't fix what ain't broken' type situation?

Comment (DJMcNab) [reply src/backend/wayland/input/keyboard/mmap.rs:10]: We're keeping the X11 backend, but we're not improving it

That is, because it mostly works, it's useful as a fallback

File: src/backend/wayland/application.rs
Comment (DJMcNab) [reply src/backend/wayland/application.rs:65]: I would like to avoid random 'Application' instances keeping the connection alive(?) without reason, which it's possible that this would do.

But I'm not tied to this particular implementation decision

File: Cargo.toml
Comment (jaredoconnell) [inline Cargo.toml:129]: Should this be grouped with the other wayland dependencies below?

How much do you need to use the lower-level wayland-client library after the switch to smithay-client-toolkit? I do understand that the toolkit is just designed to abstract some of what wayland-client offers.

File: src/backend/linux/window.rs
Comment (jaredoconnell) [inline src/backend/linux/window.rs:268]: Will this need to be re-implemented later?

**Review by jaredoconnell — APPROVED** [review]
Looks like a good start. This looks good to merge once PR #117 is merged.

File: src/backend/linux/window.rs
Comment (DJMcNab) [reply src/backend/linux/window.rs:268]: I think so - it was previously used in `create_popup`, which accepts a `crate::WindowHandle` rather than a `wayland::WindowHandle`

File: Cargo.toml
Comment (DJMcNab) [reply Cargo.toml:129]: These are just entirely unused dependencies. I've removed them

**Pull request merged**
